internal/exec/stages/files/filesystemEntries.go

```diff
@@ -174,8 +177,8 @@ func (tmp linkEntry) create(l *log.Logger, u util.Util) error {
 // mapEntriesToFilesystems builds a map of filesystems to files. If multiple
 // definitions of the same filesystem are present, only the final definition is
 // used. The directories are sorted to ensure /foo gets created before /foo/bar.
-func (s stage) getOrderedCreationList(config types.Config) []filesystemEntry {
-	entries := []filesystemEntry{}
+func (s stage) getOrderedCreationList(config types.Config) (entries []filesystemEntry, err error) {
+	entries = []filesystemEntry{}
```

Comment:
This seems unnecessary.

Reply:
It is if we want the sorting function to be able to set the return value of this function

Reply:
Interesting, didn't realize that the sorting function couldn't refer to variables defined in the function returns.